File: 約瑟夫問題(小規模).c
#include <stdio.h>

int josephus(int n, int k) {
    return n == 1 ? 0 : (josephus(n - 1, k) + k) % n;
}

int main() {
    int n, k;
    while(scanf("%d %d", &n, &k) == 2)
        printf("The last person standing is at position %d\n", josephus(n, k) + 1);
	return 0;
}

File: 約瑟夫問題(小規模).java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer st = new StreamTokenizer(br);
        StringBuilder output = new StringBuilder();
        while(st.nextToken() == StreamTokenizer.TT_NUMBER) {
            int n = (int)st.nval;
            st.nextToken();
            int k = (int)st.nval;
            output.append("The last person standing is at position ").append(josephus(n, k) + 1).append('\n');
        }
        System.out.print(output);
    }
    
    static int josephus(int n, int k) {
        return n == 1 ? 0 : (josephus(n - 1, k) + k) % n;
    }
}
